torch/quantization/fx/graph_module.py

```diff
@@ -47,8 +44,34 @@ def __deepcopy__(self, memo):
         fake_mod.__dict__ = copy.deepcopy(self.__dict__)
         return ObservedStandaloneGraphModule(fake_mod, self.graph)
 
-def mark_observed_standalone_module(module: GraphModule) -> GraphModule:
-    return ObservedStandaloneGraphModule(module, module.graph)
-
 def is_observed_standalone_module(module: Any) -> bool:
     return isinstance(module, ObservedStandaloneGraphModule)
+
+
+class QuantizedGraphModule(GraphModule):
+    def __init__(self, root: Union[torch.nn.Module, Dict[str, Any]], graph: Graph):
+        super().__init__(root, graph)
+
+    def _save_to_state_dict(self, destination, prefix, keep_vars):
```

Comment:
Why not use the hooks instead of overriding the function itself: https://github.com/pytorch/pytorch/blob/8af648354f4ae8e00576030f558ee08deb1c4acd/torch/nn/modules/module.py#L1008 ?

Reply:
sure, I can use hooks

Reply:
why there is no self in the prehook? https://github.com/pytorch/pytorch/blob/8af648354f4ae8e00576030f558ee08deb1c4acd/torch/nn/modules/module.py#L1127
I will need to modify self in this case.

Can I use hook for _save_to_state_dict but overload _load_from_state_dict?

Reply:
>  I will need to modify self in this case.

Why? The current code doesn't right?
Also modifying the module itself when it is saved sounds wrong no?

Reply:
I mean I need to modify the module when loading from state_dict

Reply:
Ho that's interesting that this is not there... I guess you're only supposed to change the state dict itself from there?
cc @jbschlosser